services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/AutoRerouteSpec.groovy

```diff
@@ -237,6 +313,54 @@ class AutoRerouteSpec extends BaseSpecification {
         }
     }
 
+    def "Flow in 'Down' status tries to reroute when connecting a new switch"() {
+        requireProfiles("virtual")
+
+        given: "With-intermediate-switch flow with one alternate path at least"
+        def flow = withIntermediateSwitchFlow(true)
+        def allPaths = db.getPaths(flow.source.datapath, flow.destination.datapath)*.path.sort { it.size() }
+        flow.maximumBandwidth = 1000
+        northboundService.addFlow(flow)
+        def flowPath = PathHelper.convert(northboundService.getFlowPath(flow.id))
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        when: "Disconnect all intermediate switches that are involved in current and alternate paths"
+        List<PathNode> disconnectedSwitches = []
+        allPaths.flatten().unique { it.switchId }.findAll {
+            !(it.switchId in [flowPath.first(), flowPath.last()]*.switchId)
+        }.each { sw ->
+            disconnectedSwitches.add(sw)
+            lockKeeperService.knockoutSwitch(sw.switchId)
+        }
+
+        then: "Flow goes to 'Down' status"
+        Wrappers.wait(discoveryTimeout + rerouteDelay + WAIT_OFFSET * 2) {
+            northboundService.getFlowStatus(flow.id).status == FlowState.DOWN
+        }
+
+        when: "Connect switches that are involved in alternate paths"
+        disconnectedSwitches.findAll { !(it.switchId in flowPath*.switchId) }.each {
+            lockKeeperService.reviveSwitch(it.switchId)
+            disconnectedSwitches.remove(it)
+        }
+
+        then: "Flow goes to 'Up' status"
+        Wrappers.wait(rerouteDelay + discoveryInterval + WAIT_OFFSET) {
+            northboundService.getFlowStatus(flow.id).status == FlowState.UP
+        }
+
+        and: "Flow was rerouted"
+        flowPath != PathHelper.convert(northboundService.getFlowPath(flow.id))
+        Wrappers.wait(WAIT_OFFSET) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
```

Comment:
You've already checked this on previous step. Don't understand why to check again

Reply:
Yes, thanks.